ydb/public/api/protos/ydb_topic.proto

```diff
@@ -796,24 +811,89 @@ message AlterConsumer {
+    // Auto scaling disabled
+    DISABLED = 0;
+    // The auto scaling may only increase the number of partitions
+    SCALE_UP = 1;
+    // The auto scaling may increase and decrease the number of partitions
+    SCALE_UP_AND_DOWN = 2;
+}
+
+message AutoScalingPartitionWriteSpeedStrategy {
+    // The time of exceeding the threshold value in seconds, after which the partition will be
+    // auto scaling.
+    // Zero value means default - 300.
+    int32 threshold_time_seconds = 1;
+    // The threshold value of the write speed to the partition as a percentage, when exceeded,
+    // the partition will be auto split.
+    // Zero value means default - 90.
+    int32 scale_up_threshold_percent = 2 [(Ydb.value) = ">= 0"];
+    // The threshold value of the write speed to the partition as a percentage, if it is not reached,
+    // the partition will be auto merged.
+    // Zero value means default - 30.
+    int32 scale_down_threshold_percent = 3 [(Ydb.value) = ">= 0"];
+}
+
+message AutoScalingSettings {
+    // Strategy of auto scaling
+    AutoScalingStrategy strategy = 1;
+    // AutoScaling partition write speed options.
+    AutoScalingPartitionWriteSpeedStrategy partition_write_speed = 2;
+}
+
+message AlterAutoScalingPartitionWriteSpeedStrategy {
+    // The time of exceeding the threshold value in seconds, after which the partition will be
+    // auto scaling.
+    // Zero value means default - 300.
+    optional int32 set_threshold_time_seconds = 1;
+    // The threshold value of the write speed to the partition as a percentage, when exceeded,
+    // the partition will be auto split.
+    // Zero value means default - 90.
+    optional int32 set_scale_up_threshold_percent = 2 [(Ydb.value) = ">= 0"];
+    // The threshold value of the write speed to the partition as a percentage, if it is not reached,
+    // the partition will be auto merged.
+    // Zero value means default - 30.
+    optional int32 set_scale_down_threshold_percent = 3 [(Ydb.value) = ">= 0"];
+}
+
+message AlterAutoScalingSettings {
+    // Strategy of auto scaling
+    optional AutoScalingStrategy set_strategy = 1;
+    // AutoScaling partition write speed options.
+    optional AlterAutoScalingPartitionWriteSpeedStrategy set_partition_write_speed = 2;
+}
+
 // Partitioning settings for topic.
 message PartitioningSettings {
     // Minimum partition count auto merge would stop working at.
     // Zero value means default - 1.
     int64 min_active_partitions = 1 [(Ydb.value) = ">= 0"];
+    // Maximum partition count auto merge would stop working at.
+    // Zero value means default - 1.
+    int64 max_active_partitions = 3 [(Ydb.value) = ">= 0"];
     // Limit for total partition count, including active (open for write) and read-only partitions.
     // Zero value means default - 100.
-    int64 partition_count_limit = 2 [(Ydb.value) = ">= 0"];
+    // Use max_active_partitions
+    int64 partition_count_limit = 2 [(Ydb.value) = ">= 0", deprecated = true];
```

Comment:
What if user has wave-like load and very often split and merge partitions? Will we want to limit total partitions count?

Reply:
Is it mean that if I have retention - a week, I have a peak of load (for example black Friday) and my one topic split to ten - then I need about 3 week for revert to one topic - as before peak of load?